tests/autodiff/long-loop-multiple.slang

```diff
@@ -0,0 +1,41 @@
+//TEST(compute):COMPARE_COMPUTE_EX:-slang -compute -shaderobj -output-using-type
+//TEST(compute, vulkan):COMPARE_COMPUTE_EX:-vk -compute -shaderobj -output-using-type
+//TEST:SIMPLE(filecheck=CHECK): -target hlsl -profile cs_5_0 -entry computeMain -line-directive-mode none
+
+//TEST_INPUT:ubuffer(data=[0], stride=4):out,name=outputBuffer
+RWStructuredBuffer<float> outputBuffer;
+
+[BackwardDifferentiable]
+float sin_series(float x, int iterations)
+{
+    float result = x;
+    float term = x;
+    [MaxIters(30)]
+    for (int i = 1; i < iterations * 10; i += 10)
```

Comment:
We can use more tests to ensure the check is at least conservative. Such as what if `i` is modified inside a loop from a branch? Can we handle the case where there are multiple `i+=k` statements in the loop, can we fold them into the same affine transform?

Reply:
> Dataflow by any other name would smell as sweet

shouldn't be a problem